modules/gdscript/gdscript_parser.cpp

```diff
@@ -7956,6 +7956,7 @@ void GDScriptParser::_check_class_level_types(ClassNode *p_class) {
 
 		_mark_line_as_safe(v.line);
 		v.data_type = _resolve_type(v.data_type, v.line);
+		v.initial_assignment->arguments[0]->set_datatype(v.data_type);
```

Comment:
Is `initial_assignment` always available? If a variable is not set this would be `nullptr` and crash, or am I missing something?

Reply:
every var declaration has initial_assignment of `OperatorNode::OP_INIT_ASSIGN` with the first argument as the identifier(always available unless parser error) and the second may or may not be `nullptr`. 
And the identifier's type was unresolved when when it created, but it has a type [fixed in my early pr](https://github.com/godotengine/godot/pull/37598/files#diff-8b4334df54a8abf5b41fe22cf292f9edR4945).
now i'm re assigning the datatype of the identifier after resolving the type.
@vnen @avencherus